packages/go_router/lib/src/delegate.dart

```diff
@@ -119,13 +129,10 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
         // Stop if there is a pageless route on top of the shell route.
         break;
       }
-
-      if (potentialCandidate.canPop()) {
-        state = walker.navigatorKey.currentState;
-      }
+      states.insert(0, potentialCandidate);
```

Comment:
you can do a reversed at the end

Reply:
Thanks, I updated the code to use reversed and iterable as return type.